src/macros/addons/MarkerWithLabel.jsx

```diff
@@ -0,0 +1,150 @@
+/* global google */
+import React from "react"
+import PropTypes from "prop-types"
+import _MarkerWithLabel from "markerwithlabel"
+import ReactDOM from "react-dom"
+
+import {
+  componentDidMount,
+  componentDidUpdate,
+  componentWillUnmount,
+  construct,
+} from "../../utils/MapChildHelper"
+
+import {
+  ANCHOR,
+  MAP,
+  MARKER_CLUSTERER,
+  MARKER_WITH_LABEL,
+} from "../../constants"
+
+export const __jscodeshiftPlaceholder__ = `{
+  "eventMapOverrides": {
+    "onDblClick": "dblclick",
+    "onDragEnd": "dragend",
+    "onDragStart": "dragstart",
+    "onMouseDown": "mousedown",
+    "onMouseOut": "mouseout",
+    "onMouseOver": "mouseover",
+    "onMouseUp": "mouseup",
+    "onRightClick": "rightclick"
+  },
+  "getInstanceFromComponent": "this.state[MARKER_WITH_LABEL]"
+}`
+
+/**
+ * @url https://developers.google.com/maps/documentation/javascript/3.exp/reference#Marker
+ */
+export class Marker extends React.PureComponent {
+  static propTypes = {
+    __jscodeshiftPlaceholder__: null,
+    /**
+     * For the 2nd argument of `MarkerCluster#addMarker`
+     * @see https://github.com/mikesaidani/marker-clusterer-plus
+     */
+    noRedraw: PropTypes.bool,
+    /**
+     * For `MarkerWithLabel`
+     * @see https://cdn.rawgit.com/googlemaps/v3-utility-library/master/markerwithlabel/src/markerwithlabel.js
+     */
+    labelClass: PropTypes.string,
+    /**
+     * For `MarkerWithLabel`
+     * @see https://cdn.rawgit.com/googlemaps/v3-utility-library/master/markerwithlabel/src/markerwithlabel.js
+     */
+    labelAnchor: PropTypes.object,
+
+    /**
+     * For `MarkerWithLabel`
+     * @see https://cdn.rawgit.com/googlemaps/v3-utility-library/master/markerwithlabel/src/markerwithlabel.js
+     */
+    labelStyle: PropTypes.object,
+  }
+
+  static contextTypes = {
+    [MAP]: PropTypes.object,
+    [MARKER_CLUSTERER]: PropTypes.object,
+  }
+
+  static childContextTypes = {
+    [ANCHOR]: PropTypes.object,
+  }
+
+  /*
+   * @url https://developers.google.com/maps/documentation/javascript/3.exp/reference#Marker
+   */
+  constructor(props, context) {
+    super(props, context)
+    const markerWithLabel = new _MarkerWithLabel(google.maps)
+    const container = document.createElement("div")
+    ReactDOM.render(this.props.children, container)
```

Comment:
Don't render in the `constructor` method. Please see  the corresponding code in `InfoBox/InfoWindow`